palace/utils/configfile.hpp

```diff
@@ -647,29 +656,13 @@ struct EigenSolverData
   bool mass_orthog = false;
 
   // Eigenvalue solver type.
-  enum class Type
-  {
-    DEFAULT,
-    SLEPC,
-    ARPACK,
-    FEAST
-  };
+  using Type = WavePortData::EigenSolverType;
```

Comment:
Seems a little surprising to have the EigenSolverType a subtype within WavePortData rather than freestanding?

Reply:
I agree I didn't love this but also wanted to be consistent that everywhere else `enum`s are defined in the scope of a `struct` which uses them. The fact that it is defined here instead of in `EigenSolverData` as it was previously is just because of ordering in the file. Is there a better solution?

Reply:
Practically it doesn't make that much of a difference ultimately given this type alias, so not a big deal. It does highlight the difficulty of that convention though, as it introduces this side coupling. I would prefer the enum outside, as at this point the notion of an `EigenSolverType` is "more common", but it's just a style thing so consistency is king here.

Reply:
Yes I would support the refactor to having all enums be described outside the structs, and we could us `using` statements in the struct scope to avoid needing naming changes to access them. That is a good idea for a later PR to make it all consistent.